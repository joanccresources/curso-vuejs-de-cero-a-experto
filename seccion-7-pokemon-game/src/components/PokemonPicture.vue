<template>
  <div class="pokemon-container">
    <img :src="imgSrc" alt="pokemon" class="img-fluid img-hidden" />
    <img
      v-if="showPokemon"
      :src="imgSrc"
      alt="pokemon"
      class="img-fluid fade-in"
    />
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    pokemonId: {
      type: Number,
      required: true,
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false,
    },
  },

  data(): { id: number } {
    return {
      id: this.pokemonId,
    };
  },

  computed: {
    imgSrc(): string {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.id}.svg`;
    },
  },
});
</script>

<style lang="scss" scoped>
.pokemon-container {
  height: 200px;
}
img {
  height: 200px;
  position: absolute;
  right: 32%;
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}
.img-hidden {
  filter: brightness(0);
}
</style>
