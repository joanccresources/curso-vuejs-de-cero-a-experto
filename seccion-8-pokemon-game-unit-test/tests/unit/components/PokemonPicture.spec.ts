import { shallowMount } from "@vue/test-utils";
import PokemonPicture from "@/components/PokemonPicture.vue";

describe("PokemonPicture", () => {
  //  npm run test:unit PokemonPicture.spec.ts -- -u
  test("debe de hacer match con el snapshoot", () => {
    // Montamos el componente y le pasamos los props
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de mostrar la imagen oculta y el pokemon 100", () => {
    const pokemonId = 100;
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId,
        showPokemon: false,
      },
    });

    const [img1, img2] = wrapper.findAll("img");
    // Si efectivamente existe la 1er imagen
    expect(img1.exists()).toBeTruthy();
    // Pasa si img2 es no definido (por las puras sabemos que hay solo uno)
    expect(img2).toBe(undefined);

    // console.log(img1.classes()); // [ 'img-fluid', 'img-hidden' ]

    // Si tiene la clase "img-hidden" pasa
    expect(img1.classes("img-hidden")).toBe(true);

    // Asi obtenemos el src de la imagen    
    expect(img1.attributes("src")).toBe(
      `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemonId}.svg`
    );
    
  });

  test("debe de mostrar el pokemon si showPokemon:true", () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: true,
      },
    });

    // Solo hay una imagen
    const img = wrapper.find("img");
    // Existe esta imagen ?
    expect(img.exists()).toBeTruthy();
    // Esta imagen tiene la clase "fade-in"?
    expect(img.classes("fade-in")).toBe(true);
  });
});
