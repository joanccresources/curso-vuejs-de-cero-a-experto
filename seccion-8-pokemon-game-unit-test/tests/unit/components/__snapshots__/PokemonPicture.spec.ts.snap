// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture debe de hacer match con el snapshoot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon" class="img-fluid img-hidden"></div>`;
